client-mr/core/src/main/java/org/apache/hadoop/mapred/SortWriteBuffer.java

```diff
@@ -341,22 +341,57 @@ public SortBufferIterator(SortWriteBuffer<K, V> sortWriteBuffer) {
       this.iterator = sortWriteBuffer.records.iterator();
     }
 
+    private byte[] fetchDataFromBuffers(int startIndex, int offset, int length) {
+      int bufferIndex = startIndex;
```

Comment:
Since startIndex, offset, length only used here, do we need thees variables? Here is extra data copy.